File: andorid-mvvm02-databinding/app/src/main/java/com/s3d/databinding/MainActivity.java
package com.s3d.databinding;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.databinding.ViewDataBinding;

import android.os.Bundle;

import com.s3d.databinding.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Car car = new Car(R.drawable.car,"BMW" ,"I3", 250);
        Driver driver = new Driver("Jack",R.drawable.driver , 33);

        ActivityMainBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        binding.setCar(car);
        binding.setDriver(driver);

    }
}

File: android-mvvm13-viewbinding-recyclerView/app/src/main/java/com/s3d/myviewbindingrv/MainActivity.kt
package com.s3d.myviewbindingrv

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.LayoutInflater
import android.widget.LinearLayout
import androidx.recyclerview.widget.LinearLayoutManager
import com.s3d.myviewbindingrv.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding :ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
    }

    override fun onResume() {
        super.onResume()

        binding.recyclerView.adapter = UserListAdapter(UserGenerator.getUsers())
        binding.recyclerView.layoutManager = LinearLayoutManager(this)

    }
}